modules/game/src/main/BSONHandlers.scala

```diff
@@ -89,14 +89,18 @@ object BSONHandlers {
       val gameVariant = Variant(r intD F.variant) | chess.variant.Standard
 
       val decoded = r.bytesO(F.huffmanPgn).map { PgnStorage.Huffman.decode(_, playedPlies) } | {
-        val clm = r.get[CastleLastMove](F.castleLastMove)
+        val clm      = r.get[CastleLastMove](F.castleLastMove)
+        val pgnMoves = PgnStorage.OldBin.decode(r bytesD F.oldPgn, playedPlies)
         PgnStorage.Decoded(
-          pgnMoves = PgnStorage.OldBin.decode(r bytesD F.oldPgn, playedPlies),
+          pgnMoves = pgnMoves,
           pieces = BinaryFormat.piece.read(r bytes F.binaryPieces, gameVariant),
           positionHashes = r.getO[chess.PositionHash](F.positionHashes) | Array.empty,
           unmovedRooks = r.getO[UnmovedRooks](F.unmovedRooks) | UnmovedRooks.default,
           lastMove = clm.lastMove,
-          castles = clm.castles
+          castles = clm.castles,
+          halfMoveClock = pgnMoves.reverse.indexWhere(san =>
+            san.contains("x") || san.headOption.exists(_.isLower)
+          ) atLeast 0
```

Comment:
The `OldBin` decoder does not provide the halfmove clock, but we can find captures and pawn moves easily.